rust/sdk/src/traits/version_tracker_step.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{
+    traits::{PollableAsyncRunType, Processable},
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use async_trait::async_trait;
+#[async_trait]
+pub trait VersionTrackerStep<T>:
+    Processable<Input = T, Output = T, RunType = PollableAsyncRunType>
+where
+    T: Send + 'static,
+{
+    // Required methods to be implemented by a tracker.
+    async fn save_processor_status(&mut self) -> Result<(), ProcessorError>;
+    fn get_next_version(&self) -> u64;
+    fn get_seen_versions(&mut self) -> &mut AHashMap<u64, TransactionContext<T>>;
+    fn get_last_success_batch(&self) -> &Option<TransactionContext<T>>;
+    fn set_next_version(&mut self, next_version: u64);
+    fn set_last_success_batch(&mut self, last_success_batch: Option<TransactionContext<T>>);
```

Comment:
My concern with all these methods is they seem pretty specific to the way this particular tracker works. What if there is another tracker that doesn't maintain next version, seen versions, last success batch, etc in memory? If they're stored elsewhere these methods will need to be async for example.

I think it might help to have an idea of what the next version tracker we write will do to motivate whether this makes sense. If it does then great, having such simple methods probably makes it easier for folks to make implementations of this trait, I just worry about flexibility.

Reply:
It's a good point but I don't see a use case there. 

The core logic will be the same (as seen in the postgres version tracker). The enhancements will be around get_start_ver, which is currently in `/utils`

Reply:
Could the in memory stuff be extracted out to something higher level then and the trait purely defines the stuff that the DB layer actually does, like save_processor_status.

Reply:
That's a good separation! That way we wouldn't have to include the getters and setters in the trait